lib/shared-method.js

```diff
@@ -404,19 +404,24 @@ function convertToBasicRemotingType(type) {
     type = type.modelName || type.name;
   }
 
-  type = String(type).toLowerCase();
+  var normalizedType = String(type).toLowerCase();
 
-  switch (type) {
+  if (normalizedType.indexOf('res:') === 0) {
+    return normalizedType;
+  }
+
+  switch (normalizedType) {
     case 'string':
     case 'number':
     case 'integer':
     case 'date':
     case 'boolean':
     case 'buffer':
     case 'object':
-    case 'file':
     case 'any':
-      return type;
+      return normalizedType;
+    case 'file':
+      return type === 'File' ? normalizedType : ('res:' + normalizedType);
```

Comment:
IMO we need to use a different condition here: only `file` should be treated as a built-in file type, everything else is a custom model (`FILE`, FilE`, etc.).